### ISSUE TITLE
nist_score vs bleu_score's SmoothingFunction.method3

### ISSUE THREAD

**Issue by goodmami** [nltk/nltk#1842]
The 3.2.5 release introduced the `nltk.translate.nist_score` module, but I was under the impression that the BLEU score using `nltk.translate.bleu_score.SmoothingFunction.method3()` for smoothing was the NIST metric, as its docstring says "NIST geometric sequence smoothing". Are these two totally different things? They do give very different scores:

```python
>>> from nltk.translate import nist_score, bleu_score
>>> hypothesis1 = ['It', 'is', 'a', 'guide', 'to', 'action', 'which','ensures', 'that', 'the', 'military', 'always','obeys', 'the', 'commands', 'of', 'the', 'party']
>>> reference1 = ['It', 'is', 'a', 'guide', 'to', 'action', 'that','ensures', 'that', 'the', 'military', 'will', 'forever','heed', 'Party', 'commands']
>>> nist_score.sentence_nist([reference1], hypothesis1)
0.05876787197888857
>>> bleu_score.sentence_bleu([reference1], hypothesis1, smoothing_function=bleu_score.SmoothingFunction().method3)
0.41180376356915777
```

The docstring for `nist_score.sentence_nist()` says it comes from `mteval-14.pl`, while the Chen & Cherry 2014 paper says their method 3 comes from `mteval-v13a.pl`. Perhaps we could just update the docstrings of the respective functions so it's more clear that the smoothing function and the NIST metric are not the same, so that people who aren't already familiar with the implementation (like me) are not confused.

**Comment by alvations** [comment]
Sadly, NIST score is still broken. Although the NIST tokenizers are working fine. https://github.com/nltk/nltk/pull/1840

But I think it's a good idea to clean up the documentation too. 

The `SmoothingFunction().method3` is a smoothing method from NIST but NIST score reports a totally different score from BLEU, it doesn't scale to the 0-0.1 range too. Smoothing only changes the precision part of the score but the NIST has a different brevity penalty and also a different way to compute the final score. 

You can take a look at the examples from the files in `nltk_data/models/wmt15_eval`, to download, `>>> nltk.download('models/wmt15_eval')`.

To summarize:

 - NIST is a different score from BLEU
 - NIST don't range from 0-1.0
 - NIST calculate n-grams precision differently from BLEU, its using some information weights (which we are still figuring out in https://github.com/nltk/nltk/pull/1840/files =( )
 - NIST smoothing in BLEU is introduced in `mteval-v*.pl` to smooth BLEU and NIST precision
 - NIST comes with its own brevity penalty different from BLEU
 - BLEU uses geometric mean of the ngram overlaps, NIST uses arithmetic mean.

**Comment by goodmami** [comment]
Thanks for the explanation. Sorry I missed #1840 when looking for discussion about this---my search criteria included `is:issue`.

I'm glad you're working on `nist_score`, because otherwise I would have continued thinking that I was computing the NIST score when I was just using NIST smoothing :)

**Comment by alvations** [comment]
Resolved. NIST was completed in #1840 a couple of month before =)
